### ISSUE TITLE
As Virologist my body was Disintegrated it disconnected me and when I reconnected it had me pick a role

### ISSUE THREAD

**Issue by Kromgar** [tgstation/-tg-station#6738]
I'm not sure how my body disintigrated admin didn't know how but somehow it caused my DC and I could pick a new role.

![image](https://cloud.githubusercontent.com/assets/6778786/5593351/5b557172-91d5-11e4-9a05-39438beaa9fb.png)

**Comment by tkdrg** [comment]
What happened right before you disconnected? Please give a detailed description.

**Comment by Kromgar** [comment]
I was in the restroom before the disconnect and the admin only told me that my body disintigrated and he didn't know how

**Comment by Kromgar** [comment]
Well I managed to get back into viro and a pile of dust and bones was there with all my shit. There was also a wizard with constructs running around so maybe I got soul sharded and that somehow caused my DC?

**Comment by tkdrg** [comment]
I investigated ingame and I still have no idea what's going on. Maybe the soul shard deleted your mob without transferring your mind. I checked runtime logs but they were already suppressed by the time this happened due to another bug.

**Comment by tkdrg** [comment]
This took way longer than it should, but, after tons of log diving, I figured it out. Here's the scoop:
You became an artificer while you were AFK. The guy who constructed you adminhelped it, and so @stickymayhem edited your artificer's ckey to that of the replacement. However, when he did that, he removed all references to your ckey from any mobs, thus crashing your client and booting you from the game. Once you logged again, you had no mob and thus were assigned a new_player.

TL;DR: an admin did an unsafe varedit while you were AFK and crashed your game. there's no bug in the code.

**Comment by Iamgoofball** [comment]
Dead issue and all but

we really need to add a failsafe to either ghost people or boot to lobby instead of disconnecting completely in situations like this

**Comment by tkdrg** [comment]
I'm pretty sure this is a byond restriction. ckeys that have no mob attached to them get instabooted, end of story. What we could do is keep people who don't have +DEBUG from editing ckey and forcing them to use a safer proc that includes ghostize() etc.

**Comment by Stickymayhem** [comment]
oops. I generally place the ckey in another mob but I must have either misspelled it or missed a step. 

It would be helpful if there was a way to simply boot someone from a mob without ckey juggling, but this was entirely my mistake and I'm sorry about that.

**Comment by tkdrg** [comment]
@Stickymayhem You can use "rudimentary transformation" in player panel to "observer" and say "no" to "delete mob?". Then you can varedit the (now-empty) ckey to the replacement. This is a lot safer. As a rule of thumb, don't edit (non-blank) ckeys unless you really know that's what you want to do.
